python/regression/PyROOT_regressiontests.py

```diff
@@ -798,6 +798,65 @@ class MyPyClass6345(cppyy.gbl.MyClass6345):
        self.assertEqual(a.bar(), 3)
        self.assertRaises(TypeError, a.bar, 0)
 
+   def test10DeepHierarchyVirtualCall(self):
+       """Virtual call resolution in deep hierarchy (C++->Py->Py)"""
+       # 6470
+
+       cppyy.gbl.gInterpreter.Declare('''
+       class CppBase6470 {
+       public:
+          virtual ~CppBase6470() {}
+          virtual int fun1() const { return 1; }
+          virtual int fun2() const { return fun1(); }
+       };
+       ''')
+
+       CppBase = cppyy.gbl.CppBase6470
+
+       class PyA(CppBase):
+          def fun1(self): return 11
+
+       class PyB(PyA):
+          def fun1(self): return 111
+
+       class PyC(PyA):
+          pass
+
+       base = CppBase()
+       a = PyA()
+       b = PyB()
+       c = PyC()
+
+       self.assertEqual(base.fun2(), 1)
+       self.assertEqual(a.fun2(), 11)
+       self.assertEqual(b.fun2(), 111)
+       self.assertEqual(c.fun2(), 11) # PyA's fun1 should be invoked
+
+   def test11MultiInheritancePyCpp(self):
+       """Multiple inheritance from Python and Cpp classes"""
+
+       class PurePy1:
+          def foo(self): return 1
+
+       class PurePy2:
+          def bar(self): return 2
+
+       cppyy.gbl.gInterpreter.Declare('''
+       class MyCppClass11 {
+       public:
+          int foo() { return 3; }
+          int bar() { return 4; }
+          virtual ~MyCppClass11() {}
+       };
+       ''')
+
+       # Multiple Python classes and just one C++ class are supported
```

Comment:
Do we also need a test for C++ -> C++ -> Python? Am I right that this is trivially handled on the C++ side?

Reply:
Yes you are right, that would be handled on the C++ side. It's the construction of the deep hierarchy on the Python side that was not working properly because the inheritance was not "chained" correctly.